### PR TITLE
BE-38 implement crawler for referaty

### PR DESCRIPTION
**Pull request by Hemii**

### CODE REVIEW

**Comment by MCFreddie777** [comment]
Please don't forget to freeze the pip packages into requirements text file.

**Review by MaliniakovaKarin — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+.vscode
+.idea
```
Comment (MCFreddie777) [inline .gitignore]: Newline :D

File: src/crawler_referaty/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+astroid==2.4.2
+autopep8==1.5.4
+colorama==0.4.4
+isort==5.6.4
+lazy-object-proxy==1.4.3
+mccabe==0.6.1
+pycodestyle==2.6.0
+pylint==2.6.0
+six==1.15.0
```
Comment (MCFreddie777) [inline src/crawler_referaty/requirements.txt]: I cannot find scrappy in the dependencies.

File: src/crawler_referaty/scraper.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import scrapy
+import json
+import os.path
+
+class ReferatySpider(scrapy.Spider):
+    name = 'referaty'
+    dirname = os.path.dirname(__file__)
+    output_file = open(os.path.join(dirname,'../../data/referaty.json'), 'a+', encoding='utf8' )
```
Comment (MCFreddie777) [inline src/crawler_referaty/scraper.py]: So it worked here, and did not in other python scripts? :trollface:

File: src/crawler_referaty/scraper.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        self.output_file.write("[\n")
+
+    def start_requests(self):
+        yield scrapy.Request(url=self.url, callback=self.parse)
+
+    def close_spider(self, spider):
+        self.output_file.write("]\n")
+        self.output_file.close()
+
+    def start_requests(self):
```
Comment (MCFreddie777) [inline src/crawler_referaty/scraper.py]: Duplicate method, please remove.

File: src/crawler_referaty/scraper.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            current_item,
+            self.output_file, 
+            ensure_ascii=False,
+            indent=""
+        )
+            
+        self.output_file.write(",\n")
+
+
+
```
Comment (MCFreddie777) [inline src/crawler_referaty/scraper.py]: Missing newline :smile:

File: src/crawler_referaty/scraper.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        json.dump(
+            current_item,
+            self.output_file, 
+            ensure_ascii=False,
+            indent=""
+        )
+            
+        self.output_file.write(",\n")
+
```
Comment (MCFreddie777) [inline src/crawler_referaty/scraper.py]: Unnecessary whitespace.

**Review by MCFreddie777 — CHANGES_REQUESTED** [review]
Reviewed, left some comments.

**Review by MCFreddie777 — APPROVED** [review]
Good work. 🐝

**Pull request merged**